code/messaging/src/main/java/com/adobe/marketing/mobile/aepcomposeui/components/SmallImageCard.kt

```diff
@@ -52,38 +68,67 @@ fun SmallImageCard(
             observer?.onEvent(UIEvent.Interact(ui, UIAction.CLICK))
         }
     ) {
-        AepRowComposable(
-            rowStyle = style.rootRowStyle
-        ) {
-            // TODO - Add image support
-            AepColumnComposable(
-                columnStyle = style.textColumnStyle
+        Box {
+            ui.getTemplate().dismissBtn?.let {
+                AepIconComposable(
+                    drawableId = it.drawableId,
+                    // todo check if we can remember this calculation so that it is not repeated for recompositions
+                    style = style.dismissButtonStyle.apply {
+                        modifier = (modifier ?: Modifier).align(style.dismissButtonAlignment)
+                    },
+                    onClick = {
+                        observer?.onEvent(UIEvent.Dismiss(ui))
+                    }
+                )
+            }
+
+            AepRowComposable(
+                rowStyle = style.rootRowStyle
             ) {
-                ui.getTemplate().title.let {
-                    AepTextComposable(
-                        model = it,
-                        textStyle = style.titleAepTextStyle
+                bitmap?.let {
+                    AepImageComposable(
+                        content = BitmapPainter(it.asImageBitmap()),
+                        style = style.imageStyle
                     )
-                }
-                ui.getTemplate().body?.let {
-                    AepTextComposable(
-                        model = it,
-                        textStyle = style.bodyAepTextStyle
+                } ?: run {
+                    Box(
+                        modifier = Modifier
+                            .size(AepUIConstants.SmallImageCard.DefaultStyle.IMAGE_WIDTH.dp)
+                            .shimmerEffect(
+                                shimmerColor1 = style.cardStyle.colors?.containerColor,
+                                shimmerColor2 = style.cardStyle.colors?.contentColor,
```

Comment:
Sorry, I missed your comment on adding the shimmer in the PR. 
I am in two minds here. I wonder if customers will always want a shimmer effect and if they do the shimmer style will vary depending on their aesthetics. Should this be optional? Or should we provide default customizable animations for loading transitions (appear with fade, shimmer etc).
Can you share a short screen capture on how this looks? We can leave this as is for the initial version - but I am leaning towards allowing them to provide customizable effects.

Reply:
Attached screenshot of what the shimmer looks like. Regarding making it optional, as per the spec, it looks like its planned for the next phase. A placeholder image can be provided until the image is downloaded from the URL
https://wiki.corp.adobe.com/display/adms/AEPImage

When we add this, we can also provide an option to in `AepImageStyle` for client side placeholder which can accept a composable as a placeholder where app developers can pass whatever animation they want. Till then we can leave it as shimmer. Does that work?

Reply:
Alternatively I can add `placeholder:  @Composable () -> Unit` right away so that client side loading indicator is supported from the first version. What do you think?

Reply:
For the initial version, can we use a simple [`fadeIn` animation](https://developer.android.com/develop/ui/compose/animation/composables-modifiers#enter-exit-transition) when the image appears and display no content until the image is downloaded? Is that an option?